lib/main.dart

```diff
@@ -1,10 +1,144 @@
+import 'dart:async';
+import 'dart:developer';
+
 import 'package:flutter/material.dart';
 import 'package:firebase_core/firebase_core.dart';
+import 'package:awesome_notifications/awesome_notifications.dart';
+
 import 'package:therabot/navigation/app.dart';
+import 'package:therabot/store/notif_provider.dart';
 
+StreamSubscription<ReceivedNotification>? receivedNotificationStream;
+StreamSubscription<ReceivedAction>? receivedActionStream;
 
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
   await Firebase.initializeApp();
+  await _createTherabotNotification();
+  _createTherabotNotificationListener();
+
   runApp(const App());
 }
+
+Future<void> _initializeNotifications(
+    AwesomeNotifications notifications) async {
+  await notifications.initialize(
+    null,
+    [
+      NotificationChannel(
+        channelGroupKey: 'therabot_prompt_group',
+        groupAlertBehavior: GroupAlertBehavior.All,
```

Comment:
I know this is test code and still in development, but I feel like it is worth refactoring this into a `notifications` directory.